addons/crm_survey/tests/test_crm_survey.py

```diff
@@ -0,0 +1,257 @@
+        self.assertTrue(last_lead_created)
+        self.assertEqual(last_lead_created.name, f"{user.display_name}'s survey results")
+
+        # Ensure that the result values are present in lead description
+        description = last_lead_created.description
+        for answer in user_inputs.user_input_line_ids:
+            self.assertIn(answer._get_answer_value(), description)
+
+        # Ensure contact, salesperson, medium, source, email and the contact name are rights
+        self.assertEqual(last_lead_created.partner_id, user.partner_id)
+        self.assertFalse(last_lead_created.user_id.id)
+        self.assertEqual(last_lead_created.medium_id.name, "Survey")
+        self.assertEqual(last_lead_created.source_id.name, survey.title)
+        self.assertEqual(last_lead_created.email_from, user.email)
+        self.assertEqual(last_lead_created.contact_name, user.partner_id.name)
+
+        # Ensure Odoobot created the lead
+        for message in last_lead_created.message_ids:
+            self.assertEqual(message.author_id.name, self.env.ref('base.user_root').name)
+
+    def test_connected_account_access_without_lead_generation_answer(self):
+        # Step 2 : Connected access + no lead generation
+        survey = self._create_lead_qualification_survey(survey_name="Not important survey")
+
+        # Account connection
+        login_password = 'survey_user'
+        user = self.env['res.users'].search([('login', '=', login_password)])
+
+        # Start page
+        self._access_start(survey)
+        user_inputs = self.env['survey.user_input'].search([('survey_id', '=', survey.id)])
+        answer_token = user_inputs.access_token
+
+        # First page
+        response = self._access_page(survey, answer_token)
+        csrf_token = self._find_csrf_token(response.text)
+        self._access_begin(survey, answer_token)
+
+        # Answers
+        with self.with_user(login_password):
+            question_ids = list(survey.question_ids)
+            self._answer_question(question_ids[0], question_ids[0].suggested_answer_ids.ids[2], answer_token, csrf_token)
+            self._answer_question(question_ids[1], question_ids[1].suggested_answer_ids.ids[2], answer_token, csrf_token)
+            self._answer_question(question_ids[2], question_ids[2].suggested_answer_ids.ids[2], answer_token, csrf_token)
+            self._answer_question(question_ids[3], question_ids[3].suggested_answer_ids.ids[2], answer_token, csrf_token)
+            self._answer_question(question_ids[4], user.email, answer_token, csrf_token)
+
+        ### Check if the last created lead was from the survey
+        last_lead_created = self.env['crm.lead'].search([], order='create_date desc', limit=1)
+        self.assertNotEqual(last_lead_created.name, f"{user.display_name}'s survey results")
+
+    def test_not_connected_account_access_with_lead_generation_answer(self):
```

Comment:
`public_user` is the usual wording :) '`logged_in`' otherwise, or more specific `survey_manager`

`test_survey_with_lead_generation_public(self)` for instance here